plugins/header_rewrite/expander.cc

```diff
@@ -113,6 +113,13 @@ VariableExpander::expand(const Resources &res)
         }
         TSHandleMLocRelease(bufp, TS_NULL_MLOC, url_loc);
       }
+    } else if (variable == "%<cque>") {
+      // The client request effective URL.
+      int url_len = 0;
+      const char *url = TSHttpTxnEffectiveUrlStringGet(res.txnp, &url_len);
```

Comment:
Yes, I did not notice that `TSHttpTxnEffectiveUrlStringGet` always calls `HTTPHdr::url_string_get` with a null first parameter. This causes new memory to be allocated for the resulting url string, and this must be freed. This is in contrast to calls to `TSHttpHdrMethodGet` and `TSUrlPathGet` in the two cases directly above in which the strings returned point to memory that is managed elsewhere.